File: VB/CommonDbContext/ViewModels/SchedulingViewModel.vb
Imports System.Collections.Generic
Imports System.Data.Entity
Imports System.Linq
Imports DevExpress.Mvvm.Native
Imports DevExpress.Xpf.Scheduling
Imports DXSample.Data

Namespace DXSample.ViewModels
	Public Class SchedulingViewModel
		Private dbContext As SchedulingContext
		Public Sub New()
			dbContext = New SchedulingContext()
			Resources = dbContext.ResourceEntities.ToList()
		End Sub
		Public Overridable Property Resources() As List(Of ResourceEntity)
		Public Sub CreateSourceObject(ByVal args As CreateSourceObjectEventArgs)
			If args.ItemType = ItemType.AppointmentItem Then
				args.Instance = New AppointmentEntity()
			End If
		End Sub
		Public Sub FetchAppointments(ByVal args As FetchDataEventArgs)
			Dim res = dbContext.AppointmentEntities.Where(args.GetFetchExpression(Of AppointmentEntity)())
			args.AsyncResult = QueryableExtensions.ToArrayAsync(Of Object)(res)
		End Sub
		Public Sub ProcessChanges(ByVal args As AppointmentCRUDEventArgs)
			dbContext.AppointmentEntities.AddRange(args.AddToSource.Select(Function(x) CType(x.SourceObject, AppointmentEntity)))
			dbContext.AppointmentEntities.RemoveRange(args.DeleteFromSource.Select(Function(x) CType(x.SourceObject, AppointmentEntity)))
			dbContext.SaveChanges()
		End Sub
	End Class
End Namespace

File: CS/CommonDbContext/ViewModels/SchedulingViewModel.cs
using System.Collections.Generic;
using System.Data.Entity;
using System.Linq;
using DevExpress.Mvvm.Native;
using DevExpress.Xpf.Scheduling;
using DXSample.Data;

namespace DXSample.ViewModels {
    public class SchedulingViewModel {
        SchedulingContext dbContext;
        public SchedulingViewModel() {
            dbContext = new SchedulingContext();
            Resources = dbContext.ResourceEntities.ToList();
        }
        public virtual List<ResourceEntity> Resources { get; set; }
        public void CreateSourceObject(CreateSourceObjectEventArgs args) {
            if(args.ItemType == ItemType.AppointmentItem)
                args.Instance = new AppointmentEntity();
        }
        public void FetchAppointments(FetchDataEventArgs args) {
            args.AsyncResult = dbContext.AppointmentEntities.
                Where(args.GetFetchExpression<AppointmentEntity>()).
                ToArrayAsync<object>();
        }
        public void ProcessChanges(AppointmentCRUDEventArgs args) {
            dbContext.AppointmentEntities.AddRange(args.AddToSource.Select(x => (AppointmentEntity)x.SourceObject));
            dbContext.AppointmentEntities.RemoveRange(args.DeleteFromSource.Select(x => (AppointmentEntity)x.SourceObject));
            dbContext.SaveChanges();
        }
    }
}